topic-operator/src/test/java/io/strimzi/operator/topic/KafkaStreamsTopicStoreClusterTest.java

```diff
@@ -0,0 +1,112 @@
+        store1.create(topic)
+                .onComplete(context.succeeding())
+
+                // Read the topic
+                .compose(v -> store2.read(new TopicName(topicName)))
+                .onComplete(context.succeeding(readTopic -> context.verify(() -> {
+                    // assert topics equal
+                    assertThat(readTopic.getTopicName(), is(topic.getTopicName()));
+                    assertThat(readTopic.getNumPartitions(), is(topic.getNumPartitions()));
+                    assertThat(readTopic.getNumReplicas(), is(topic.getNumReplicas()));
+                    assertThat(readTopic.getConfig(), is(topic.getConfig()));
+                })))
+
+                // try to create it again: assert an error
+                .compose(v -> store2.create(topic))
+                .onComplete(context.failing(e -> context.verify(() -> {
+                    assertThat(e, instanceOf(TopicStore.EntityExistsException.class));
+                    failedCreateCompleted.complete();
+                })));
+
+        Topic updatedTopic = new Topic.Builder(topic)
+                .withNumPartitions(3)
+                .withConfigEntry("fruit", "apple")
+                .build();
+
+        failedCreateCompleted.future()
+                .compose(v -> {
+                    // update my_topic
+                    return store2.update(updatedTopic);
+                })
+                .onComplete(context.succeeding())
+
+                // re-read it and assert equal
+                .compose(v -> store1.read(new TopicName(topicName)))
+                .onComplete(context.succeeding(rereadTopic -> context.verify(() -> {
+                    // assert topics equal
+                    assertThat(rereadTopic.getTopicName(), is(updatedTopic.getTopicName()));
+                    assertThat(rereadTopic.getNumPartitions(), is(updatedTopic.getNumPartitions()));
+                    assertThat(rereadTopic.getNumReplicas(), is(updatedTopic.getNumReplicas()));
+                    assertThat(rereadTopic.getConfig(), is(updatedTopic.getConfig()));
+                })))
+
+                // delete it
+                .compose(v -> store2.delete(updatedTopic.getTopicName()))
+                .onComplete(context.succeeding())
+
+                // assert we can't read it again
+                .compose(v -> store1.read(new TopicName(topicName)))
+                .onComplete(context.succeeding(deletedTopic -> context.verify(() ->
+                        assertThat(deletedTopic, is(nullValue()))))
+                )
+
+                // delete it again: assert an error
+                .compose(v -> store1.delete(updatedTopic.getTopicName()))
+                .onComplete(context.failing(e -> context.verify(() -> {
+                    assertThat(e, instanceOf(TopicStore.NoSuchEntityExistsException.class));
+                    async.flag();
+                })));
+    }
+
+    @AfterEach
+    public void teardown(VertxTestContext context) {
```

Comment:
This teardown doesn't do anything.. am I missing something here?

Reply:
It does that Checkpoint::flag ... whatever that does , as I've copied this from previous test.

Reply:
Ah, this is not needed anymore, the checkpont needs to be flagged or it delays a test from finishing, since you aren't waiting for any actual teardown, this is redundant. 

This is sort of analogous to doing

future = new Future()
future.complete()

If you're waiting on nothing it is just not needed.

Reply:
So I can remove this as well in "KafkaStreamsTopicStoreTest" ?

Reply:
Removed ...